Duplicate enum name (simple + parametric) causes assertion failure

## Compiler version

Scala 3.8.1

## Minimized code
```scala
object Test {
  enum E {
    case K0
  }
  enum E:
    case K0(v0: Long)
}
```

## Output (click arrow to expand)

<details>
java.lang.AssertionError: assertion failed: class K0 has non-class parent: TypeRef(NoPrefix,val <none>)
Phase: typer
at Namer.defDefSig -> inferredResultType -> baseClasses

</details>

## Expectation

The compiler should emit a normal error about the duplicate definition of `enum E` (and/or `K0`) rather than crashing with an internal `AssertionError`.

## Note
Found using my Scala 3 grammar model and [DepTyCheck](https://github.com/buzden/deptycheck)
Also found via my tools: https://github.com/scala/scala3/issues/25248, https://github.com/scala/scala3/issues/25246, https://github.com/scala/scala3/issues/25244

--- a/tests/neg/duplicate-enum.scala
+++ b/tests/neg/duplicate-enum.scala
@@ -1,0 +1,7 @@
+object Test {
+  enum E { // error
+    case K0
+  }
+  enum E: // error
+    case K0(v0: Long)
+}

